app/lib/flag-route.js

```diff
@@ -1,16 +1,29 @@
-import Route from '@ember/routing/route';
+const ROUTE_LIX_ENABLED_KEY = '__route_lix_is_enabled';
 
 function generateRouteHandler(owner, key, treatmentRouteName) {
   const treatmentRoute = owner.lookup(`route:${treatmentRouteName}`);
   const flagService = owner.lookup('service:flags');
   return {
+    set(target, prop, value) {
+      const flagIsControl = !Reflect.get(target, ROUTE_LIX_ENABLED_KEY);
+      if (prop === 'controller') {
+        // Set the other controller too, because _internalReset assumes controller is set once route is setup
+        if (flagIsControl) {
+          Reflect.set(treatmentRoute, prop, owner.lookup(`controller:${treatmentRouteName}`));
```

Comment:
should it return here?

Reply:
If flag is not enabled, should the controller _not_ be treatmentRouteName? It should be whatever [`controllerName`](https://api.emberjs.com/ember/3.20/classes/Route/properties?anchor=controllerName) is set to, or default to same as routeName.

Reply:
The naming convention is a it concerning to me. Routes need to have `old-` and `new-`, but templates and controllers only have `new-`. But that should be a doc thing for people who specifically opt into this technique.

Reply:
This code isn't choosing a controller. It is setting both controllers on the respective Route objects. There are two controllers, one for the control route and one for the enabled route object.

Reply:
There's no requirement for the naming, besides the existing requirement that the class, controller, and template files are the same.

Reply:
Point taken though that the file names used in this repo are confusing. will fix that.

Reply:
would it be better to move content `route/profile/old-component.js` to `route/profile/component`, instead of importing? In real world, the `component.js` route would be the existing working route before adding `new-component.js`. The example looks like we need to add a third `route` file, but I think we just need two to let it work